### PR TITLE
Change the order in the signatures of array APIs

### PR DESCRIPTION
**Pull request by adithyaov**

### CODE REVIEW

**Comment by harendra-kumar** [comment]
docspec Ci needs fixing.

**Comment by harendra-kumar** [comment]
Are any of the changed APIs released?

File: src/Streamly/Internal/Data/Array/Foreign.hs [hunk 1]
```diff
@@ -431,8 +431,8 @@ getSlicesFromLen from len =
 --
 -- @since 0.8.0
 {-# INLINE getIndex #-}
-getIndex :: forall a. Storable a => Array a -> Int -> Maybe a
-getIndex arr i =
+getIndex :: forall a. Storable a => Int -> Array a -> Maybe a
+getIndex i arr =
```
Comment (adithyaov) [inline src/Streamly/Internal/Data/Array/Foreign.hs:435]: This is released!

Comment (adithyaov) [reply src/Streamly/Internal/Data/Array/Foreign.hs:435]: Reverting this change so we can merge this.

File: src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs [hunk 2]
```diff
@@ -2249,8 +2249,8 @@ cast arr =
 --
 -- /Pre-release/
 --
-asPtrUnsafe :: Array a -> (Ptr b -> IO c) -> IO c
-asPtrUnsafe Array{..} act = do
+asPtrUnsafe :: (Ptr b -> IO c) -> Array a -> IO c
+asPtrUnsafe act Array{..} = do
```
[the remark this answers is not in the document]
Comment (adithyaov) [reply src/Streamly/Internal/Data/Array/Foreign/Mut/Type.hs:2253]: Ah, we were!

**Review by harendra-kumar — APPROVED** [review]

**Pull request merged**